crypto/bn/asm/loongarch-mont.pl

```diff
@@ -0,0 +1,333 @@
+#! /usr/bin/env perl
+# Copyright 2010-2018 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+#
+# ====================================================================
+# Written by Song Ding <songding@loongson.cn> for the OpenSSL
+# project. The module is, however, dual licensed under OpenSSL and
+# CRYPTOGAMS licenses depending on where you obtain it. For further
+# details see http://www.openssl.org/~appro/cryptogams/.
+# ====================================================================
+
+# December 2021
+#
+# This module doesn't present direct interest for OpenSSL, because it
+# doesn't provide better performance for longer keys, at least not on
+# in-order-execution cores. While 512-bit RSA sign operations can be
+# 65.7% faster in 64-bit mode, 1024-bit ones are 24.6% faster, and
+# 2048-bit ones are only 10.4% faster, the code improve rsa512 *verify*
+# benchmark by 20.4%, rsa1024 one - by 10.9%:-)
+# All comparisons are against bn_mul_mont-free assembler.
```

Comment:
Fairly weird comment here - if the improvement for contemporary RSA key lengths is negligible, is it worth it?

Reply:
These figures were taken on a 2.5GHz machine, the performance will be better on a 2.3GHz machine . Also it can improve more than 100% that compared with "no-asm"